config/http_config.go

```diff
@@ -309,6 +309,9 @@ type HTTPClientConfig struct {
 	// The omitempty flag is not set, because it would be hidden from the
 	// marshalled configuration when set to false.
 	EnableHTTP2 bool `yaml:"enable_http2" json:"enable_http2"`
+	// Host optionally overrides the Host header to send.
+	// If empty, the host from the URL will be used.
+	Host string `yaml:"host" json:"host"`
```

Comment:
If this is optional, it should be `omitempty` I think.

```suggestion
	Host string `yaml:"host,omitempty" json:"host,omitempty"`
```